### ISSUE TITLE
[youtube] Downloading youtube videos that have an empty live chat fails

### ISSUE THREAD

**Issue by rebane2001** [blackjack4494/yt-dlc#161]
<!--

######################################################################
  WARNING!
  IGNORING THE FOLLOWING TEMPLATE WILL RESULT IN ISSUE CLOSED AS INCOMPLETE
######################################################################

-->


## Checklist

<!--
Carefully read and work through this check list in order to prevent the most common mistakes and misuse of youtube-dlc:
- First of, make sure you are using the latest version of youtube-dlc. Run `youtube-dlc --version` and ensure your version is 2020.10.31. If it's not, see https://github.com/blackjack4494/yt-dlc on how to update. Issues with outdated version will be REJECTED.
- Make sure that all provided video/audio/playlist URLs (if any) are alive and playable in a browser.
- Make sure that all URLs and arguments with special characters are properly quoted or escaped as explained in https://github.com/blackjack4494/yt-dlc.
- Search the bugtracker for similar issues: https://github.com/blackjack4494/yt-dlc. DO NOT post duplicates.
- Read bugs section in FAQ: https://github.com/blackjack4494/yt-dlc
- Finally, put x into all relevant boxes (like this [x])
-->

- [x] I'm reporting a broken site support issue
- [x] I've verified that I'm running youtube-dlc version **2020.10.31**
- [x] I've checked that all provided URLs are alive and playable in a browser
- [x] I've checked that all URLs and arguments with special characters are properly quoted or escaped
- [x] I've searched the bugtracker for similar bug reports including closed ones
- [x] I've read bugs section in FAQ


## Verbose log

<!--
Provide the complete verbose output of youtube-dlc that clearly demonstrates the problem.
Add the `-v` flag to your command line you run youtube-dlc with (`youtube-dlc -v <your command line>`), copy the WHOLE output and insert it below. It should look similar to this:
 [debug] System config: []
 [debug] User config: []
 [debug] Command-line args: [u'-v', u'http://www.youtube.com/watch?v=BaW_jenozKcj']
 [debug] Encodings: locale cp1251, fs mbcs, out cp866, pref cp1251
 [debug] youtube-dlc version 2020.10.31
 [debug] Python version 2.7.11 - Windows-2003Server-5.2.3790-SP2
 [debug] exe versions: ffmpeg N-75573-g1d0487f, ffprobe N-75573-g1d0487f, rtmpdump 2.4
 [debug] Proxy map: {}
 <more lines>
-->

```
[youtube] 05uY49DjC7U: Downloading webpage                                                                                                                                                                                                                                                                                [youtube] 05uY49DjC7U: Downloading MPD manifest                                                                                                                                                                                                                                                                           [info] Writing video description to: m_20201110/Evershade (UC0FPGo7lj4gqauBnjptR2_Q)/20191130 - [Music] Through The Clouds - 05uY49DjC7U.description                                                                                                                                                                      WARNING: There are no annotations to write.
[info] Writing video subtitles to: m_20201110/Evershade (UC0FPGo7lj4gqauBnjptR2_Q)/20191130 - [Music] Through The Clouds - 05uY49DjC7U.live_chat.json
[debug] Invoking downloader on None
[youtube_live_chat_replay] Downloading live chat
[youtube_live_chat_replay] Total fragments: unknown (live)
[download] Destination: m_20201110/Evershade (UC0FPGo7lj4gqauBnjptR2_Q)/20191130 - [Music] Through The Clouds - 05uY49DjC7U.live_chat.json
[download] 474.91KiB at 890.90KiB/s (00:00)ERROR: 'liveChatReplayContinuationData'
Traceback (most recent call last):
  File "/usr/local/bin/youtube-dl/youtube_dlc/YoutubeDL.py", line 846, in extract_info
    return self.process_ie_result(ie_result, download, extra_info)
  File "/usr/local/bin/youtube-dl/youtube_dlc/YoutubeDL.py", line 901, in process_ie_result
    return self.process_video_result(ie_result, download=download)
  File "/usr/local/bin/youtube-dl/youtube_dlc/YoutubeDL.py", line 1696, in process_video_result
    self.process_info(new_info)
  File "/usr/local/bin/youtube-dl/youtube_dlc/YoutubeDL.py", line 1894, in process_info
    dl(sub_filename, sub_info, subtitle=True)
  File "/usr/local/bin/youtube-dl/youtube_dlc/YoutubeDL.py", line 1866, in dl
    return fd.download(name, info, subtitle)
  File "/usr/local/bin/youtube-dl/youtube_dlc/downloader/common.py", line 373, in download
    return self.real_download(filename, info_dict)
  File "/usr/local/bin/youtube-dl/youtube_dlc/downloader/youtube_live_chat.py", line 85, in real_download
    continuation_id = live_chat_continuation['continuations'][0]['liveChatReplayContinuationData']['continuation']
KeyError: 'liveChatReplayContinuationData'
```


## Description

<!--
Provide an explanation of your issue in an arbitrary form. Please make sure the description is worded well enough to be understood, see https://github.com/ytdl-org/youtube-dlc#is-the-description-of-the-issue-itself-sufficient. Provide any additional information, suggested solution and as much context and examples as possible.
If work on your issue requires account credentials please provide them or explain how one can obtain them.
-->

If a video with a live chat (such as a premiere) is being downloaded, but the live chat has 0 messages, the download fails.
Here is an example of a video affected by this issue: https://www.youtube.com/watch?v=05uY49DjC7U

**Comment by rebane2001** [comment]
I have remedied this issue by changing 
```Py
continuation_id = live_chat_continuation['continuations'][0]['liveChatReplayContinuationData']['continuation']
```
to
```Py
try:
    continuation_id = live_chat_continuation['continuations'][0]['liveChatReplayContinuationData']['continuation']
except:
    continuation_id = None
```

I will not be creating this as a PR as I believe this is more of a band-aid than a proper fix, but I'll leave this code here in case anybody needs to patch their own setup ASAP.

**Comment by pukkandan** [comment]
```python
continuation_id = live_chat_continuation['continuations'][0].get('liveChatReplayContinuationData', {}).get('continuation', None)
```

would fix the issue. But before I make a PR, I donot know whether the other keys could fail to exist. 
```python
continuation_id = data['contents']['twoColumnWatchNextResults']['conversationBar']['liveChatRenderer']['continuations'][0]['reloadContinuationData']['continuation']

live_chat_continuation = data['continuationContents']['liveChatContinuation']

continuation_id = live_chat_continuation['continuations'][0]['liveChatReplayContinuationData']['continuation']
```

These 3 lines are potential points of failure. Should I convert all the keys to get? Or are they confirmed to always exist?

**Comment by rebane2001** [comment]
During my testing, it seems like my patch handles all the cases that I've encountered (and I've encountered a lot of them), so I do not think the other potential points of failure are affected by this issue. However, it might make sense to look into them as well.

**Comment by rebane2001** [comment]
Seems like a fix was put in place in 8263104fe4f7aed96a1cc92be6b58cc219de876e
